src/main/java/com/limechain/client/FullNode.java

```diff
@@ -58,8 +65,62 @@ public void start() {
         AppBean.getBean(WarpSyncMachine.class).start();
     }
 
-    void initializeDatabaseStorageTrie() {
-        TrieStructure<Pair<Optional<byte[]>, byte[]>> trieStructure;
+    /**
+     * Builds a list of {@link InsertTrieNode} objects representing the nodes in a trie structure.
+     * Each trie node is constructed with its storage value, merkle value, children's merkle values,
+     * and partial key nibbles.
+     *
+     * @param trieStructure The trie structure containing the nodes. This structure should be
+     *                      a {@link TrieStructure} with user data of type {@link Pair<Optional<byte[]>, byte[]>},
+     *                      where the first element of the pair is an optional storage value (byte array),
+     *                      and the second element is the merkle value (byte array).
+     * @return A list of {@link InsertTrieNode} objects representing the nodes in the given trie structure.
+     * @throws IllegalStateException if the user data in the trie structure is empty or null, which
+     *                               indicates an invalid state for the trie nodes.
+     */
+    public List<InsertTrieNode> buildTrieNodesIterator(
+            TrieStructure<Pair<Optional<byte[]>, byte[]>> trieStructure) {
```

Comment:
As far as I gather, we're mixing an `Optional<byte[]>` for the storage value and then a pure `byte[]` for the merkle value, which we check for `null` to determine whether the field is present or not... Any particular reason for that decision? 😄